src/ethereum/ethEngine.js

```diff
@@ -166,41 +175,64 @@ export class EthereumEngine extends CurrencyEngine {
     }
   }
 
-  async checkAccountFetch (tk: string, url: string) {
+  updateBalance (tk: string, balance: string) {
+    if (typeof this.walletLocalData.totalBalances[tk] === 'undefined') {
```

Comment:
Please consider `x == null` instead of `typeof x === 'undefined'`. It's the same thing, but a lot shorter. If you really mean exactly "`undefined` and not `null`", you can just say that, like `x === undefined`.

The `typeof` thing is only useful in exactly one case: When you have a top-level variable like `window` that may or may not even exist at all, then you really do need `typeof`, since touching a non-existing variable any other way is an error.

Reply:
This was just moving old existing code that admittedly could have just been
`    if (this.walletLocalData.totalBalances[tk]) {`
 I'll make note for future refactors